docs/src/contributing_apub_api_outline.md

```diff
@@ -1,380 +1,589 @@
+|---|---|---|
+| `attributedTo` | yes | ID of the user who created the comment |
+| `to` | yes | Community where the comment was made |
+| `content` | yes | The comment text |
+| `inReplyTo` | yes | IDs of the post where this comment was made, and the parent comment. If this is a top-level comment, `inReplyTo` only contains the post |
+| `published` | no | Datetime when the comment was created |
+| `updated` | no | Datetime when the comment was edited (not present if it was never edited) |
+
+### Private Message
+
+```json
 {
-  "@context": "https://www.w3.org/ns/activitystreams",
-  "type": "Delete",
-  "actor": id,
-  "object": comment_id, or post_id
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://enterprise.lemmy.ml/private_message/34",
+    "type": "Note",
+    "attributedTo": "https://enterprise.lemmy.ml/u/picard",
+    "to": "https://voyager.lemmy.ml/u/janeway",
+    "content": "test",
+    "published": "2020-10-08T19:10:46.542820+00:00",
+    "updated": "2020-10-08T20:13:52.547156+00:00",
 }
 ```
 
-#### [Update](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-update)
-```
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `attributedTo` | ID of the user who created this private message |
+| `to` | ID of the recipient |
+| `content` | yes | The text of the private message |
+| `published` | no | Datetime when the message was created |
+| `updated` | no | Datetime when the message was edited (not present if it was never edited) |
+
+## Activities
+
+TODO: need to check that mandatory fields are correct
+
+TODO: inboxes should probably parse the `to` field instead of whatever they are doing now
+
+TODO: for activities sent from user to community, `cc` should simply be the community, not the followers
+
+### Common Fields
+
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `actor` | yes | The user or community which is sending the activity |
+| `to` | no | Who this activity is being sent to, can be a specific actor or public |
+| `cc` | no | Generally the list of community followers, or users mentioned in a comment. Completely ignored by Lemmy |
+| `object` | yes | The thing that is being changed |
+
+### Follow
+
+Sent by: User
+
+Sent to: Community
+
+```json
 {
-  "@context": "https://www.w3.org/ns/activitystreams",
-  "type": "Create",
-  "actor": id,
-  "object": comment_id, or post_id
-  TODO fields.
+    "@context": "https://www.w3.org/ns/activitystreams",
+    "id": "https://enterprise.lemmy.ml/activities/follow/2e4784b7-4edf-4fa1-a352-674d5d5f8891",
+    "type": "Follow",
+    "actor": "https://enterprise.lemmy.ml/u/picard",
+    "object": "https://ds9.lemmy.ml/c/main"
 }
 ```
-#### [Read](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-read)
-```
+
+| Field Name | Mandatory | Description |
+|---|---|---|
+| `actor` | yes | The user that is sending the follow request |
+| `object` | yes | The community to be followed |
+
+TODO: shouldn't the Follow also have `to`?
```

Comment:
Fixed in https://yerbamate.dev/LemmyNet/lemmy/commit/2ad60379e41b653cc4c841d024e2d9cdf7884f69